insonmnia/hub/server.go

```diff
@@ -861,3 +880,48 @@ func (h *Hub) deleteTask(taskID string) error {
 
 	return nil
 }
+
+func (h *Hub) initLocatorClient() error {
+	conn, err := grpc.Dial(
+		h.locatorEndpoint,
+		grpc.WithInsecure(),
+		grpc.WithTimeout(5*time.Second),
+		grpc.WithDecompressor(grpc.NewGZIPDecompressor()),
+		grpc.WithCompressor(grpc.NewGZIPCompressor()))
+	if err != nil {
+		return err
+	}
+
+	h.locatorClient = pb.NewLocatorClient(conn)
+	return nil
+}
+
+func (h *Hub) startLocatorAnnouncer() {
+	tk := time.NewTicker(h.locatorPeriod)
+	defer tk.Stop()
+
+	h.announceAddress(h.ctx)
+
+	for {
+		select {
+		case <-tk.C:
+			h.announceAddress(h.ctx)
```

Comment:
Do we really need to do it in cycle? Maybe we can track connection from hub to locator